pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -981,59 +986,102 @@ func (a *Operator) updateNamespaceList(op *operatorsv1.OperatorGroup) ([]string,
 	return namespaceList, nil
 }
 
-func (a *Operator) ensureOpGroupClusterRole(op *operatorsv1.OperatorGroup, suffix string, apis cache.APISet) error {
-	clusterRole := &rbacv1.ClusterRole{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   strings.Join([]string{op.GetName(), suffix}, "-"),
-			Labels: map[string]string{install.OLMManagedLabelKey: install.OLMManagedLabelValue},
-		},
+func (a *Operator) stableRand(op *operatorsv1.OperatorGroup) (string, error) {
+	key := fmt.Sprintf("%s/%s", op.GetNamespace(), op.GetName())
+	h := fnv.New64a()
+	_, err := h.Write([]byte(key))
+	if err != nil {
+		return "", err
 	}
-	var selectors []metav1.LabelSelector
-	for api := range apis {
-		aggregationLabel, err := aggregationLabelFromAPIKey(api, suffix)
-		if err != nil {
-			return err
-		}
-		selectors = append(selectors, metav1.LabelSelector{
-			MatchLabels: map[string]string{
-				aggregationLabel: "true",
-			},
-		})
+	rnd := rand.NewSource(int64(h.Sum64()))
+	const alphabet = "abcdefghijklmnopqrstuvwxyz0123456789"
+
+	b := make([]byte, 7)
+	for i := range b {
+		b[i] = alphabet[rnd.Int63()%int64(len(alphabet))]
 	}
-	if len(selectors) > 0 {
-		clusterRole.AggregationRule = &rbacv1.AggregationRule{
-			ClusterRoleSelectors: selectors,
-		}
+	return string(b), nil
+}
+
+func (a *Operator) getClusterRoleName(op *operatorsv1.OperatorGroup, roleType string) (string, error) {
+	roleSuffix, err := a.stableRand(op)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("olm.operatorgroup.%s-%s", roleType, roleSuffix), nil
```

Comment:
e.g. https://pkg.go.dev/hash/crc32

Reply:
if we don't care so much about leaking the namespace/og names (because they are already labeled in the ClusterRole resource anyway) - we could just remove the whole hash thing in favor of using genName instead of name in the ClusterRole CR. We can just have something like: io.operator-framework.operator-group.<namespace>.<name>.<type>-XXXXX (where k8s can handle the XXXXX).

Reply:
I'd be in favor of removing the "hashing". Who came up with the idea/had the concern? They should be consulted.

Reply:
Now reconsidering the removal... as adding in the namepsace and name of the OG could generate a name that's too long. So, my preference would be to change the hash such that it can be calculated outside of golang.
Although, I suppose, a search with selectors might be faster to implement, script-wise.

Reply:
should be good now ^^